tests/test_build_linkcheck.py

```diff
@@ -575,3 +575,21 @@ def test_limit_rate_bails_out_after_waiting_max_time(app):
                                               rate_limits)
     next_check = worker.limit_rate(FakeResponse())
     assert next_check is None
+
+
+class ConnectionResetHandler(http.server.BaseHTTPRequestHandler):
+    def do_HEAD(self):
+        raise requests.ConnectionError
```

Comment:
The web server runs in a separate thread. I believe the reason the test works is that the unhandled exception causes the connection to be closed, but the raised exception does not propagate to the linkcheck builder thread.

Instead, be explicit about what the HTTP server is doing: closing the connection.
```suggestion
        self.connection.close()
```